### DISCUSSION TITLE
any way to get kubeconfig using sdkV2?

### DISCUSSION THREAD

**Discussion by Horiodino** [aws/aws-sdk-go-v2#2684] [Q&A] [answered]
for kubeconfig we use k8s go pkgs and for that sigs are using sdkV1  auth type `*session.Session` and we are using sdkV2 `aws.Config`, is there any other way, as the k8s-sigs aws authenticator is still using v1.

**Comment by RanVaknin** [comment, marked as answer]
Hi @Horiodino ,

Thanks for reaching out. I assume you are talking about [k8s-sigs/aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator)? 

Since that package is what pulls the SDK and not vice versa, I suggest opening an issue on their github asking them to support v2, especially since v1 is on the road for deprecation. 

My only experience with k8 is through EKS which abstracts all the auth part through IRSA. I'm not sure about your use case but if you can use EKS to manage your k8 infrastructure it would circumvent the need to manage authentication through a third party tool. With IRSA your EKS cluster has its auth layer that is abstracted and automatically refreshes the EKS token needed to make requests via the SDK.

I also found this https://aws-controllers-k8s.github.io/community/docs/community/overview/ , which lets you define AWS resources directly from k8.

Thanks,
Ran~

**Comment by Horiodino** [comment]
thank you :+1:
